modules/ROOT/pages/tuning-batch-processing.adoc

```diff
@@ -17,5 +19,44 @@ You can configure this property for the batch job. If you believe that in your p
 <batch:job jobName="test-batch" blockSize="${batch.block.size}">
 ----
 
+== Max Concurrency
+
+The maximum number of threads for processing a Batch scope defaults to 2 times the number of cores that the JVM detects. On fractional core deployments, such as CloudHub and Anypoint Runtime Fabric, the JVM detects a _non-fractional_ number of cores. You can manually increase or lower the automated concurrency setting. The Max Concurrency setting enables you to set an upper limit on the number of threads to use for a batch job instance that, for example, has insufficient records to warrant more threads.
+
+The follow example overrides the automated `maxConcurrency` default to a setting that is loaded from a property:
+
+[source,xml,linenums]
+----
+<batch:job jobName="an-example-batch" maxConcurrency="${batch.block.maxConcurrency}">
+----
+
+== Batch Aggregators
+
+One million round trips across a network, whether on single or multiple threads, impacts performance. Communication with SaaS systems can easily consume the quota of API calls. Instead of processing records one by one, you can improve performance by using the Batch Aggregator scope to perform bulk operations on groups of records. The scope uses an Aggregator Size setting to divide the records into groups. When determining the number of records to process in bulk, you must take into account the maximum number of records permitted by operations on the system with which the communication takes place. It is important to conduct performance testing on your Aggregator Size setting _and on lower values_. Your logic must handle any number of records up to and including the value of your setting because the number of records that the system receives can be lower than your specified value. For example, if there are 120 records, and your Aggregator Size setting is 50, the last submission is 20 records.
+
+The following example shows a Batch Step scope with a Batch Aggregator scope that is set to a fixed size of `50`:
+
+[source,xml,linenums]
+----
+<batch:step name="Batch_Step_PushToXYZ" >
+  <batch:aggregator doc:name="Batch Aggregator" size="50">
+     <!-- code for an array of 50 (or fewer) records -->
+  </batch:aggregator>
+</batch:step>
+----
+
+If the aggregator contains an operation that streams data, you can activate the scope's Streaming option  (`streaming="true"`) instead of using an Aggregator Size setting (`size="50"`).
```

Comment:
```suggestion
If the aggregator contains an operation that streams data, you can activate the scope's Streaming option (`streaming="true"`) instead of using an Aggregator Size setting (`size="50"`).
```